### PR TITLE
Define Glossary

### LINKED ISSUE #1724: Add terminology glossary to to the docs
source is #745

please also have a look at https://github.com/woodpecker-ci/woodpecker/issues/745#issuecomment-1313401427

### PR DESCRIPTION
**Pull request by 6543**
based on https://github.com/woodpecker-ci/woodpecker/issues/745#issuecomment-1313401427

close  #1724

### CODE REVIEW

File: docs/docs/92-development/09-terminology.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- **Event**: Triggers the execution of a pipeline, such as a forge event, manual interaction in the user interface, cron job, or command-line interface.
+- **Commit**: A defined state of the code, usually associated with a version control system like Git.
+- **Matrix**: A configuration option that allows the execution of workflows for each value in the matrix.
+- **Service**: A service is a step that is executed from the start of a workflow until its end. It can be accessed by name via the network from other steps within the same workflow.
+- **Plugins**: Plugins are extensions that provide pre-defined actions or commands for a step in a workflow. They can be configured via settings.
+- **Container**: A lightweight and isolated environment where commands are executed in.
+- **YAML File**: A file format used to define and configure workflows.
+- **Dependency**: Workflows can depend on each other, and if possible, they are executed in parallel.
+- **Status**: Status refers to the outcome of a step or workflow after it has been executed, determined by the internal command exit code.
+
```
Comment (anbraten) [inline docs/docs/92-development/09-terminology.md:22]: additional terms: server, agent, cli, backend

File: docs/docs/92-development/09-terminology.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## Glossary
+
+- **Woodpecker**: An open-source tool that executes pipelines on your code.
+- **Pipeline**: A sequence of workflows that are executed on the code. Pipelines are triggered by events.
+- **Workflow**: A sequence of steps and services that are executed as part of a pipeline. Workflows are represented by YAML files. Each workflow has its own isolated network.
+- **Steps**: Individual commands, actions or tasks within a workflow.
+- **Code**: Refers to the files tracked by the version control system used by the forge.
+- **Repos**: Short for repositories, these are storage locations where code is stored.
+- **Forge**: The hosting platform or service where the repositories are hosted.
+- **Workspace**: A separate environment or area where a workflow is executed in isolation.
```
Comment (anbraten) [inline docs/docs/92-development/09-terminology.md:12]:
```suggestion
- **Workspace**: A folder shared between all steps of a workflow containing the repository and all the generated data from previous steps.
```

File: docs/docs/92-development/09-terminology.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- **Event**: Triggers the execution of a pipeline, such as a forge event, manual interaction in the user interface, cron job, or command-line interface.
+- **Commit**: A defined state of the code, usually associated with a version control system like Git.
+- **Matrix**: A configuration option that allows the execution of workflows for each value in the matrix.
+- **Service**: A service is a step that is executed from the start of a workflow until its end. It can be accessed by name via the network from other steps within the same workflow.
+- **Plugins**: Plugins are extensions that provide pre-defined actions or commands for a step in a workflow. They can be configured via settings.
```
Comment (anbraten) [inline docs/docs/92-development/09-terminology.md:17]: Maybe cross linking to the plugins .md file would be nice

File: docs/docs/92-development/09-terminology.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- **Container**: A lightweight and isolated environment where commands are executed in.
+- **YAML File**: A file format used to define and configure workflows.
+- **Dependency**: Workflows can depend on each other, and if possible, they are executed in parallel.
+- **Status**: Status refers to the outcome of a step or workflow after it has been executed, determined by the internal command exit code.
```
Comment (anbraten) [inline docs/docs/92-development/09-terminology.md:21]:
```suggestion
- **Status**: Status refers to the outcome of a step or workflow after it has been executed, determined by the internal command exit code. At the end of a workflow its status it is send to the forge.
```

File: docs/docs/92-development/09-terminology.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Terminology
+
+## Glossary
+
+- **Woodpecker**: An open-source tool that executes pipelines on your code.
```
[the remark this answers is not in the document]
Comment (6543) [reply docs/docs/92-development/09-terminology.md:5]: well "woodpecker ci" is the project ... and woodpecker the tool i would say

File: docs/docs/92-development/09-terminology.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+# Terminology
+
+## Glossary
+
+- **Woodpecker CI**: The project name around Woodpecker.
```
Comment (6543) [inline docs/docs/92-development/09-terminology.md:5]: @qwerty287 do you think that's more accurate?

**Comment by anbraten** [comment]
What do you think about moving this to the user docs. I guess it would be quite nice if users would read that to for example be able to ask much better questions in the chat, issues.

**Comment by 6543** [comment]
@qwerty287 if you have a specific place in mind just push a commit to move it there ... i dont have a specific place in mid.

**Comment by 6543** [comment]
@anbraten moved, hope it's in a better spot now

**Comment by anbraten** [comment]
@6543 I had a short look around. And liked how the following pages placed it right after the `Getting started` area.
- https://docs.gocd.org/current/introduction/concepts_in_go.html
- https://buildkite.com/docs/pipelines/glossary

So between getting started and pipeline syntax, might be a good place:

image

In addition I checked the search and would suggest we change the terms to headings so they are more important in the search and the user can use easily navigate them and then use the links you have added to navigate to the corresponding pages after getting a short introduction of that term.

Extending to something like those concepts & philosophy pages could be nice in the future as well:
- https://circleci.com/docs/concepts/
- https://tekton.dev/docs/concepts/overview/
- https://kraken.ci/docs/philosophy

Also really nice is how argo structed their docs in a tranining / like a course way. So you learn step by step more advanced concepts of the tool:
- https://argoproj.github.io/argo-workflows/walk-through/

**Pull request merged**